networkx/algorithms/hierarchy.py

```diff
@@ -42,6 +42,9 @@ def flow_hierarchy(G, weight=None):
        DOI: 10.1002/cplx.20368
        http://web.mit.edu/~cmagee/www/documents/28-DetectingEvolvingPatterns_FlowHierarchy.pdf
     """
+    # corner case: G has no edges
+    if G.size() == 0:
```

Comment:
```suggestion
    if nx.is_empty(G):
```
here, using `G.is_empty` might be more efficient because we won't be iterating over all the edges.

Reply:
Nice -- I bet this idiom could be used in other places in the codebase too.